daemon/api_test.go

```diff
@@ -4058,6 +4058,15 @@ func (s *apiSuite) testDisconnect(c *check.C, plugSnap, plugName, slotSnap, slot
 	_, err := repo.Connect(connRef, nil, nil, nil)
 	c.Assert(err, check.IsNil)
 
+	st := d.overlord.State()
+	st.Lock()
+	st.Set("conns", map[string]interface{}{
+		"consumer:plug producer:slot": map[string]interface{}{
+			"interface": "test",
+		},
+	})
```

Comment:
That's because `doDisconnect` handler needs access to `conns` now to make a copy of the connection for undo; and the connection needs to exist there or otherwise I report an error.